docs/deployment/k8s/index.mdx

```diff
@@ -1,31 +1,176 @@
+  hostLabels:
+    some-label: value
+  # Which wasmCloud version to use
+  version: 0.82.0
+  # The name of a secret in the same namespace that provides the required secrets.
+  secretName: cluster-secrets
+  # Enable the following to run the wasmCloud hosts as a DaemonSet
+  #daemonset: true
+  # The name of the image pull secret to use with wasmCloud hosts so that they
+  # can authenticate to a private registry to pull components.
+  # registryCredentialsSecret: my-registry-secret
+```
 
-Cosmonic has open sourced a basic connector that enables bridging services running in Kubernetes with services running in wasmCloud.
+The CRD requires a Kubernetes Secret with the following keys:
 
-The interface, provider, and component can be found at [on Github under `cosmonic/kubernetes-applier`][k8s-applier].
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: my-wasmcloud-cluster
+data:
+  # You can generate this with wash:
+  # `wash keys gen cluster`
+  WASMCLOUD_CLUSTER_SEED: <seed>
+  # Only required if using a NATS creds file
+  # nats.creds: <base64 encoded creds file>
+```
+:::info[Secrets required]
+The operator will fail to provision the wasmCloud Deployment if any of these secrets are missing.
+:::
 
-Check out the project's [documentation][k8s-applier-readme] for a complete example architecture and instructions on how to run it.
+### Image pull secrets
 
-[k8s-applier]: https://github.com/cosmonic/kubernetes-applier
-[k8s-applier-readme]: https://github.com/cosmonic/kubernetes-applier/tree/main/service-applier#readme
+You can also specify an image pull secret to use use with the wasmCloud hosts so that they can pull components from a private registry. This secret needs to be in the same namespace as the WasmCloudHostConfig CRD and must be a
+`kubernetes.io/dockerconfigjson` type secret. See the [Kubernetes documentation](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#registry-secret-existing-credentials) for more information on how to provision that secret.
+
+Once it is created, you can reference an image pull secret in the `WasmCloudHostConfig` CRD by
+setting the `registryCredentialsSecret` field to the name of the secret.
+
+### Deploying the operator
+
+A wasmCloud cluster requires a few things to run:
+
+- A NATS cluster with Jetstream enabled
+- WADM connected to the NATS cluster in order to support applications
+
+If you are running locally, you can use the following commands to start a
+NATS cluster and WADM in your Kubernetes cluster.
+
+#### Running NATS
+
+Use the upstream NATS Helm chart to start a cluster with the following
+values.yaml file:
+
+```yaml
+config:
+  cluster:
+    enabled: true
+    replicas: 3
+  leafnodes:
+    enabled: true
+  jetstream:
+    enabled: true
+    fileStore:
+      pvc:
+        size: 10Gi
+    merge:
+      domain: default
+```
+
+```shell
+helm upgrade --install -f values.yaml nats-cluster nats/nats

```

Comment:
Should probably add a prerequisites section at the top including a k8s cluster, appropriate access creds to said cluster, and `helm` installed and configured. It might be stating the obvious but that's probably not bad.